### ISSUE TITLE
sendKeys() throws Exception "invalid argument: 'text' must be a string" with ChromeDriver

### ISSUE THREAD

**Issue by drjayvee** [php-webdriver/php-webdriver#787]
I upgraded Ubuntu, which bumped libzip to >= 1.6. This forced me to upgrade php-webdriver from 1.7.1 to 1.8.2. Now I'm getting the following Exception for _every_ use of `RemoteWebElement::sendKeys()`:
```
Facebook\WebDriver\Exception\InvalidArgumentException: invalid argument: 'text' must be a string
  (Session info: chrome=81.0.4044.122)

~/repo/vendor/php-webdriver/webdriver/lib/Exception/WebDriverException.php:99
~/repo/vendor/php-webdriver/webdriver/lib/Remote/HttpCommandExecutor.php:370
~/repo/vendor/php-webdriver/webdriver/lib/Remote/RemoteWebDriver.php:590
~/repo/vendor/php-webdriver/webdriver/lib/Remote/RemoteExecuteMethod.php:27
~/repo/vendor/php-webdriver/webdriver/lib/Remote/RemoteWebElement.php:364
```
Here's the relevant log from chromedriver:
```
[1588772853.088][INFO]: [f52ff79e0048d0ff03848f31eb0baee4] COMMAND FindElement {
   "using": "css selector",
   "value": "input[type=\"text\"]"
}
[1588772853.089][INFO]: Waiting for pending navigations...
[1588772853.090][INFO]: Done waiting for pending navigations. Status: ok
[1588772853.116][INFO]: Waiting for pending navigations...
[1588772853.120][INFO]: Done waiting for pending navigations. Status: ok
[1588772853.120][INFO]: [f52ff79e0048d0ff03848f31eb0baee4] RESPONSE FindElement {
   "element-6066-11e4-a52e-4f735466cecf": "f50bef34-f085-49a4-a525-539c0dabb7bb"
}
[1588772853.123][INFO]: [f52ff79e0048d0ff03848f31eb0baee4] COMMAND TypeElement {
   "id": "f50bef34-f085-49a4-a525-539c0dabb7bb",
   "value": [ "superUser" ]
}
[1588772853.124][INFO]: Waiting for pending navigations...
[1588772853.125][INFO]: Done waiting for pending navigations. Status: ok
[1588772853.132][INFO]: Waiting for pending navigations...
[1588772853.134][INFO]: Done waiting for pending navigations. Status: ok
[1588772853.135][INFO]: [f52ff79e0048d0ff03848f31eb0baee4] RESPONSE TypeElement ERROR invalid argument: 'text' must be a string
  (Session info: chrome=81.0.4044.122)
[1588772853.137][INFO]: [f52ff79e0048d0ff03848f31eb0baee4] COMMAND Quit {

}
```
The PHP test code is trivial, so I won't reproduce it here.

I'm creating a web driver instance like so:
```php
$this->webDriver = RemoteWebDriver::create(
	SELENIUM_SERVER_URL,
	(new ChromeOptions())->addArguments([
		'disable-features=AvoidFlashBetweenNavigation,PaintHolding',
	])->toCapabilities()
)
```

Does this have anything to do with the experimental W3C protocol?

### Details
* Php-webdriver version: 1.8.2
* PHP version: 7.3.17
* Chromium && Chromedriver version: 81.0.4044.122
* Operating system: Ubuntu Server 18.04

**Comment by OndraM** [comment]
Hi @drjayvee, 
I cannot reproduce the issue, could you please:

- try to [disable W3C protocol](https://github.com/php-webdriver/php-webdriver/pull/674#issue-339395010) to see if there is any difference (however - it should work with or without W3C protocol)
- try to remove the whole second parameter of `RemoteWebDriver::create()` (capabilities)
- try to isolate the issue and provide some minimal reproducible case? The minimal code which could be used to reproduce the issue (including crating webdriver instance and calling the sendKeys method)

Thanks a lot!

**Comment by drjayvee** [comment]
Good morning @OndraM,

Thanks so much for getting back to me so quickly.

Disabling W3c protocol solves the issue for me!

I'd be glad to help to find the actual problem, so I did some more digging. I removed the capabilities like so:
```php
$this->webDriver = RemoteWebDriver::create(
	SELENIUM_SERVER_URL
);
// or
$this->webDriver = RemoteWebDriver::create(
	SELENIUM_SERVER_URL,
	DesiredCapabilities::chrome()
);
```
That results in the following Exception:
```
Facebook\WebDriver\Exception\InvalidArgumentException: invalid argument: entry 0 of 'firstMatch' must be a JSON object

~/repo/vendor/php-webdriver/webdriver/lib/Exception/WebDriverException.php:99
~/repo/vendor/php-webdriver/webdriver/lib/Remote/HttpCommandExecutor.php:370
~/repo/vendor/php-webdriver/webdriver/lib/Remote/RemoteWebDriver.php:136
~/repo/tehst.php
```

I will (try to) create a minimal test case.

**Comment by andrewnicols** [comment]
Hi @drjayvee ,

The error is correct in that webdriver is sending the wrong arguments:
```
[1588772853.123][INFO]: [f52ff79e0048d0ff03848f31eb0baee4] COMMAND TypeElement {
   "id": "f50bef34-f085-49a4-a525-539c0dabb7bb",
   "value": [ "superUser" ]
}
```
In W3C we no longer use `value`, but instead `text`. For some reason you're still seeing WebDriver try to use the non-W3C command.

Do you have a sample script?

I've tried the following and it all works without issue on master:
```
<?php

namespace Facebook\WebDriver;

use Facebook\WebDriver\Remote\DesiredCapabilities;
use Facebook\WebDriver\Remote\RemoteWebDriver;
use Facebook\WebDriver\Chrome\ChromeOptions;

require_once('vendor/autoload.php');

define('SELENIUM_SERVER_URL', 'http://localhost:4444/wd/hub');

$driver = RemoteWebDriver::create(
    SELENIUM_SERVER_URL,
    (new ChromeOptions())->addArguments([
        'disable-features=AvoidFlashBetweenNavigation,PaintHolding',
    ])->toCapabilities()
);

// navigate to Selenium page on Wikipedia
$driver->get('http://localhost/test.html');

// write 'PHP' in the search box
$input = $driver->findElement(WebDriverBy::cssSelector('input[type="text"]'));

// find search input element
$input->sendKeys('PHP');
```

I can also verify the inability to use `create()` without a second arg, but we should raise a separate issue for that.

**Comment by OndraM** [comment]
Thanks for the information. Minimal test case would really be helpful!

Also - make sure you use latest and also compatible Chromedriver & Chrome version:
```
chromedriver --version
chrome --version # or chromium --version
```

Latest stable version of Chrome and Chromedriver is currently 81.0.4044.138. You can download latest chromedriver here: https://chromedriver.chromium.org/

**Comment by drjayvee** [comment]
I haven't had the time yet to create a test (but I will), but I have a hunch about what might be going wrong.

@andrewnicols, your example uses
```php
define('SELENIUM_SERVER_URL', 'http://localhost:4444/wd/hub');
```
From this I gather that you're using selenium server. My test code actually connects directly to chromedriver (`http://localhost:9515`) for simplicity. Could that make a difference?

I promise to dig in and
 * provide a minimal test case
 * test with the latest version of chromium/chromedriver
 * use selenium server
It might have to wait until Monday though, as I'm actually trying to get a new release out the door.

**Comment by andrewnicols** [comment]
@drjayvee 
I've just confirmed going direct against chromedriver and a URL of `http://localhost:9515`

Chromedriver version: 81.0.4044.69
Chrome version: 81.0.4044.138

All working here with the script above and updated URL.

**Comment by drjayvee** [comment]
Turns out this whole issue is bogus. ☹️

We extend `RemoteWebElement` to add some sugar methods, and I didn't see that the constructor got a new argument `$isW3cCompliant`. Therefore, all instances had that property set to `false`, even though the actual driver expected W3C-compatible calls.

Sorry to waste everybody's time, and thank you so very much for diving into this.

At the very least we got another actual issue out of this (`create()` without 2nd arg), which I will report in a separate issue.
